### PR TITLE
freetype: support compiling with stage2 (-fno-stage1)

### PR DESCRIPTION
**Pull request by PiergiorgioZagaria**
- [X] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right.

Two things I wanted to notice are, utils_pkg (see https://github.com/hexops/mach/issues/180#issuecomment-1175993008) and test_option, `@embedFile` can't import from `../` so we need to use `addOption` in the build file. Also we can't `@embedFile` in build.zig because when running build.zig for the first time upstream/ hasn't been downloaded yet.

### CODE REVIEW

**Review by alichraghi — APPROVED** [review]

File: freetype/build.zig [hunk 1]
```diff
@@ -67,11 +67,20 @@ pub fn build(b: *std.build.Builder) !void {
...
+    main_tests.addPackage(utils_pkg);
+
     main_tests.addPackage(pkg);
     link(b, main_tests, .{ .freetype = .{
         .ft_config_path = "./test/ft",
         .brotli = true,
     } });
+    var font_option = b.addOptions();
+
+    font_option.addOption([]const u8, "font", try std.fs.cwd().readFileAlloc(b.allocator, "upstream/assets/FiraSans-Regular.ttf", std.math.maxInt(u32)));
```
Comment (iddev5) [inline freetype/build.zig:81]: You can just set ``exe.main_pkg_path`` which will solve the @embedFile issue

**Review by iddev5 — APPROVED** [review]

File: freetype/build.zig (hunk 1 above)
Comment (PiergiorgioZagaria) [reply freetype/build.zig:81]: So I add `main_tests.main_pkg_path = "upstream/assets/";` in build.zig and use `@embedFile("FiraSans-Regular.ttf")`?
It doesn't work so I don't think that's what you meant

Comment (iddev5) [reply freetype/build.zig:81]: I meant setting ``main_pkg_path`` to thisDir() and then @embedFile("../upstream...")

Comment (PiergiorgioZagaria) [reply freetype/build.zig:81]: Ah yeah, that was it thanks

**Review by slimsag — APPROVED** [review]

**Pull request merged**